### PR TITLE
Add detect Avalonia xaml code

### PR DESCRIPTION
**Pull request by rstm-sf**
Hello!

## Description
Add detect [Avalonia](https://github.com/AvaloniaUI/Avalonia) xaml code. Avalonia is a [popular](https://github.com/search?q=avalonia) framework. And recently it was [accepted](https://github.com/AvaloniaUI/Avalonia/pull/4093) to associate it with a new extension.

See also: https://github.com/AvaloniaUI/Avalonia/issues/4102

## Checklist:

- [x] **I am associating a language with a new file extension.**
  - [x] The new extension is used in hundreds of repositories on GitHub.com
    - Search results for each extension:
      
      - https://github.com/search?q=extension%3Aaxaml+NOT+nothack
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - https://github.com/search?q=extension%3Aaxaml+NOT+nothack
    - Sample license(s):
  - [ ] I have included a change to the heuristics to distinguish my language from others using the same extension.

- [ ] **I am adding a new language.**
  - [ ] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      
      -  https://github.com/search?utf8=%E2%9C%93&type=Code&ref=searchresults&q=extension%3AFOOBAR+KEYWORDS+NOT+nothack
  - [ ] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - [URL to each sample source, if applicable]
    - Sample license(s):
  
  - [ ] I have included a syntax highlighting grammar: https://github-lightshow.herokuapp.com/
  - [ ] I have updated the heuristics to distinguish my language from others using the same extension.

- [ ] **I am fixing a misclassified language**
  - [ ] I have included a new sample for the misclassified language:
    - Sample source(s):
      - [URL to each sample source, if applicable]
    - Sample license(s):
  - [ ] I have included a change to the heuristics to distinguish my language from others using the same extension.

- [ ] **I am changing the source of a syntax highlighting grammar**
  
  - Old: https://github-lightshow.herokuapp.com/
  - New: https://github-lightshow.herokuapp.com/

- [ ] **I am updating a grammar submodule**
  

- [ ] **I am adding new or changing current functionality**
  
  - [ ] I have added or updated the tests for the new or changed functionality.

- [ ] **I am changing the color associated with a language**
  
  - [ ] I have obtained agreement from the wider language community on this color change.
    - [URL to public discussion]
    - [Optional: URL to official branding guidelines for the language]

### CODE REVIEW

**Comment by lildude** [comment]
Usage on GitHub.com is waaay too low for this to be considered for inclusion as it doesn't meet the "hundreds of repositories" requirement. There are only 340 files with that extension spread across 77 unique user/repos.

>  * I have included a real-world usage sample for all extensions added in this PR:

Whoops, looks like you missed this part too.

**Comment by rstm-sf** [comment]
> Usage on GitHub.com is waaay too low for this to be considered for inclusion as it doesn't meet the "hundreds of repositories" requirement. There are only 340 files with that extension spread across 77 unique user/repos.

But repos using xaml extension are already [enough](https://github.com/search?l=C%23&q=avalonia&type=Repositories). The axaml has already been added to the extension for Visual Studio https://github.com/AvaloniaUI/AvaloniaVS/pull/161. That is, after the next release of Avalonia 0.10, the number of unique repositories should increase (note that these unique repos were created using nightly builds or previews).

Or is it better to ask for an addition after the quantity exceeds the quantity of 100 pieces (because this will be mostly used by those using VS now)? Sorry, I dont know. It seems to me that this will happen soon :)



> Whoops, looks like you missed this part too.

Sorry, it looks like I misunderstood what was in the template. Just also couldn't find [examples for xaml](https://github.com/github/linguist/tree/master/samples/XML)

**Comment by lildude** [comment]
> But repos using xaml extension are already enough.

That search result is showing repos with `avalonia` in their name. That is not the same thing as checking time popularity of the extension as [this search](https://github.com/search?q=extension%3Aaxaml+NOT+nothack&type=Code) does. It's the search you should have updated in the original template to show the usage.

We require a usage of "hundreds of repositories" for a particular extension and use 200 unique `user/repo` as the starting point.

File: samples/XML/NDepends_Example.ndproj [hunk 1]
```diff
@@ -45,6 +45,7 @@ let sourceFilesPathsToDiscard = (
         ".g.cs",        // Popular pattern to name generated files.
         ".g.vb",
         ".xaml",        // notmycode WPF xaml code
+        ".axaml",       // notmycode Avalonia xaml code
```
Comment (lildude) [inline samples/XML/NDepends_Example.ndproj:48]: You should not be modifying this file.

Comment (rstm-sf) [reply samples/XML/NDepends_Example.ndproj:48]: Sorry I didn't know

**Comment by rstm-sf** [comment]
> We require a usage of "hundreds of repositories" for a particular extension and use 200 unique `user/repo` as the starting point.

Ok let's leave this for later. It looks like this question was opened too early

Sorry for bother

**Pull request closed without merging**